src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -133,8 +155,19 @@ cdef class _AioCall:
                 self._waiter_call = None
                 raise Exception("Error with grpc_call_start_batch {}".format(call_status))
 
-            await self._waiter_call
-
+            try:
+                await self._waiter_call
+            except asyncio.CancelledError:
+                if cancel_status:
+                    details = str_to_bytes(cancel_status.details)
+                    self._references.append(details)
```

Comment:
I'm not sure here. Do we need to keep this `details` in reference? Does the `<char *>` conversion copy the object, or it just accessing the low level data structure in `PyStr`?

Reply:
Under the hood the

```python
details = str_to_bytes(cancel_status.details)
c_details = <char *>details
```
Basically initializes a Python object and gets the pointer to the internal buffer by using `PyBytes_AsString` [1].

Without creating an artificial reference the object created could be deallocated by the GC and so the pointer won't point to any valid memory area.

[1] https://docs.python.org/3/c-api/bytes.html#c.PyBytes_AsString